ethcore/src/client/client.rs

```diff
@@ -255,14 +255,9 @@ impl Importer {
 		// could be retracted in import `k+1`. This is why to understand if after all inserts
 		// the block is enacted or retracted we iterate over all routes and at the end final state
 		// will be in the hashmap
-		let map = import_results.iter().fold(HashMap::new(), |mut map, route| {
-			for hash in &route.enacted {
-				map.insert(hash.clone(), true);
-			}
-			for hash in &route.retracted {
-				map.insert(hash.clone(), false);
-			}
-			map
+		let map = import_results.iter().flat_map(|route| {
+			route.enacted.iter().map(|h| (*h, true))
+				.chain(route.retracted.iter().map(|h| (*h, false))).collect::<Vec<_>>()
```

Comment:
is collecting to a vector actually necessary here? and the purpose of using a `HashMap` was to make sure the most recent enacted/retracted value holds. The changes here seem as though you can get multiple entries for the same hash (= bad)